optimum/habana/transformers/generation/utils.py

```diff
@@ -87,6 +89,18 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
         return self.cur_step >= self.max_steps
 
 
+def _get_stopping_criteria(
+    self, generation_config: GaudiGenerationConfig, stopping_criteria: Optional[StoppingCriteriaList]
+) -> StoppingCriteriaList:
+    criteria = StoppingCriteriaList()
+    if generation_config.max_length is not None:
+        criteria.append(StaticMaxLengthCriteria(generation_config.max_length))
```

Comment:
We should rather have
```suggestion
    if generation_config.max_length is not None:
        if generation_config.static_shapes:
            criteria.append(StaticMaxLengthCriteria(generation_config.max_length))
        else:
            max_position_embeddings = getattr(self.config, "max_position_embeddings", None)
            criteria.append(
                MaxLengthCriteria(
                    max_length=generation_config.max_length,
                    max_position_embeddings=max_position_embeddings,
                )
            )
```
to be able to manage the non-static-shape case.

Reply:
Will do the modifications.